File: sancreports_v0001/sancreports_v0001/report/custom_purchase_recipt/custom_purchase_recipt.js
frappe.query_reports["Custom Purchase recipt"] = {
    filters: [
        {
            fieldname: "from_date",
            label: "From Date",
            fieldtype: "Date",
        },
        {
            fieldname: "to_date",
            label: "To Date",
            fieldtype: "Date",
        },
        {
            fieldname: "supplier",
            label: "Supplier",
            fieldtype: "Link",
            options: "Supplier",
        },
    ],

    formatter: function (value, row, column, data, default_formatter) {
        value = default_formatter(value, row, column, data);

        // 🔹 GRN USD → Item-wise GRN drill-down
        if (column.fieldname === "grn_total_usd" && data) {
            return `
                <a href="javascript:void(0)"
                   class="grn-drilldown"
                   data-pr="${data.purchase_receipt}">
                   ${value}
                </a>
            `;
        }

        // 🔹 LCV → Item-wise LCV drill-down
        if (column.fieldname === "lcv" && data && data.lcv > 0) {
            return `
                <a href="javascript:void(0)"
                   class="lcv-drilldown"
                   data-pr="${data.purchase_receipt}">
                   ${value}
                </a>
            `;
        }

        // Highlight total landed cost
        if (column.fieldname === "total_landed_cost") {
            return `<b>${value}</b>`;
        }

        return value;
    },

    onload: function () {

        // 🔹 GRN ITEM POPUP
        $(document).on("click", ".grn-drilldown", function () {
            let pr = $(this).data("pr");

            frappe.call({
                method: "sancreports_v0001.sancreports_v0001.report.custom_purchase_recipt.custom_purchase_recipt.get_item_wise_grn",
                args: { purchase_receipt: pr },
                callback: function (r) {
                    let rows = r.message.map(d => `
                        <tr>
                            <td>${d.item_code}</td>
                            <td>${d.item_name || ""}</td>
                            <td align="right">${d.qty}</td>
                            <td align="right">${d.rate}</td>
                            <td align="right">${d.amount}</td>
                        </tr>
                    `).join("");

                    frappe.msgprint({
                        title: `Item-wise GRN (${pr})`,
                        message: `
                            <table class="table table-bordered">
                                <thead>
                                    <tr>
                                        <th>Item Code</th>
                                        <th>Item Name</th>
                                        <th>Qty</th>
                                        <th>Rate (USD)</th>
                                        <th>Amount (USD)</th>
                                    </tr>
                                </thead>
                                <tbody>${rows}</tbody>
                            </table>
                        `,
                        wide: true
                    });
                }
            });
        });

        // 🔹 LCV ITEM POPUP
        $(document).on("click", ".lcv-drilldown", function () {
            let pr = $(this).data("pr");

            frappe.call({
                method: "sancreports_v0001.sancreports_v0001.report.custom_purchase_recipt.custom_purchase_recipt.get_item_wise_lcv",
                args: { purchase_receipt: pr },
                callback: function (r) {
                    let rows = r.message.map(d => `
                        <tr>
                            <td>${d.item_code}</td>
                            <td>${d.item_name || ""}</td>
                            <td align="right">${d.qty}</td>
                            <td align="right">${d.applicable_charges}</td>
                        </tr>
                    `).join("");

                    frappe.msgprint({
                        title: `Item-wise LCV (${pr})`,
                        message: `
                            <table class="table table-bordered">
                                <thead>
                                    <tr>
                                        <th>Item Code</th>
                                        <th>Item Name</th>
                                        <th>Qty</th>
                                        <th>Applicable Charges</th>
                                    </tr>
                                </thead>
                                <tbody>${rows}</tbody>
                            </table>
                        `,
                        wide: true
                    });
                }
            });
        });
    }
};

File: sancreports_v0001/sancreports_v0001/report/custom_purchase_recipt/custom_purchase_recipt.py
import frappe


def execute(filters=None):
    return get_columns(), get_data(filters)


def get_columns():
    return [
        {
            "label": "Purchase Receipt",
            "fieldname": "purchase_receipt",
            "fieldtype": "Link",
            "options": "Purchase Receipt",
            "width": 170,
        },
        {
            "label": "Posting Date",
            "fieldname": "posting_date",
            "fieldtype": "Date",
            "width": 120,
        },
        {
            "label": "Supplier",
            "fieldname": "supplier",
            "fieldtype": "Data",
            "width": 220,
        },
        {
            "label": "GRN Total (INR)",
            "fieldname": "grn_total",
            "fieldtype": "Currency",
            "width": 160,
        },
        {
            "label": "LCV",
            "fieldname": "lcv",
            "fieldtype": "Currency",
            "width": 140,
        },
        {
            "label": "Total Landed Cost",
            "fieldname": "total_landed_cost",
            "fieldtype": "Currency",
            "width": 180,
        },
    ]


def get_data(filters):
    conditions = ""
    values = {}

    if filters.get("from_date"):
        conditions += " AND pr.posting_date >= %(from_date)s"
        values["from_date"] = filters.get("from_date")

    if filters.get("to_date"):
        conditions += " AND pr.posting_date <= %(to_date)s"
        values["to_date"] = filters.get("to_date")

    if filters.get("supplier"):
        conditions += " AND pr.supplier = %(supplier)s"
        values["supplier"] = filters.get("supplier")

    query = f"""
        SELECT
            pr.name AS purchase_receipt,
            pr.posting_date,
            pr.supplier_name AS supplier,

            pr.base_grand_total AS grn_total,

            -- ✅ LCV from Landed Cost Item (CONFIRMED)
            COALESCE(SUM(lci.applicable_charges), 0) AS lcv,

            -- ✅ Total Landed Cost
            pr.base_grand_total + COALESCE(SUM(lci.applicable_charges), 0)
                AS total_landed_cost

        FROM
            `tabPurchase Receipt` pr

        LEFT JOIN
            `tabLanded Cost Item` lci
            ON lci.receipt_document = pr.name
            AND lci.docstatus = 1

        WHERE
            pr.docstatus = 1
            {conditions}

        GROUP BY
            pr.name,
            pr.posting_date,
            pr.supplier_name,
            pr.base_grand_total

        ORDER BY
            pr.posting_date DESC
    """

    return frappe.db.sql(query, values, as_dict=True)
